src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java

```diff
@@ -169,6 +175,21 @@ public JSONArray getBuildCauses(String className) throws IOException, ClassNotFo
         return result;
     }
 
+    private boolean isClassOrSubclass(Class<?> clazz, String classNameToCheck) {
+        String causeClassName = clazz.getName();
+        if (causeClassName.equals("java.lang.Object")) {
+            return false;
```

Comment:
`return causeClassName.equals(classNameToCheck)`  incase the user enters "java.lang.Object")

Reply:
And also to exit the recursion...

Reply:
`Object.class.getSuperclass` [will return null](https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#getSuperclass--), so maybe easier to use that as the base case and handle `Object` with the next `if` just like other classes.